util/str_utils.py

```diff
@@ -0,0 +1,26 @@
+import string
+
+PUNCTUATION_TRANS = str.maketrans(string.punctuation, ' ' * len(string.punctuation))
+WHITESPACE_TRANS = str.maketrans(string.whitespace, ' ' * len(string.whitespace))
+
+
+def preprocess_text(text):
+    """Method for pre-processing the given response text.
+    It:
+    * changes to lowercase
+    * removes trailing and leading spaces
+    * removes all punctuations
+    * removes double spaces
+
+    :param text: the text to be cleaned
+    :return: cleaned text
+    """
+    text = text.strip().lower()
```

Comment:
I would move the strip before whitespace removing, but after the translation - if the comment ends up with punctuation or whitespace, it will be stripped immidiately.

Reply:
Fixed in 625e2c754e39226e83ff1895cda4e16c728fe7fe